ycsb-mongodb/mongodb/src/main/java/com/yahoo/ycsb/db/MongoDbClient.java

```diff
@@ -233,6 +338,42 @@ private static AutoEncryptionSettings generateEncryptionSettings(String url, Boo
         return autoEncryptionSettings;
     }
 
+    private static ArrayList<Long> parseCommaSeparatedIntegers(String toParse, int outputListSize, long defaultValue) {
+        if (toParse.trim().isEmpty()) {
+            return new ArrayList<Long>();
+        }
+        // -1 for limit implies that any trailing empty strings are included in the output array
+        String[] values = toParse.split(",", -1);
+
+        if (outputListSize < 0) {
+            outputListSize = values.length;
+        }
+        ArrayList<Long> parsedValues = new ArrayList<Long>(Collections.nCopies(outputListSize, defaultValue));
+        int limit = Math.min(values.length, outputListSize);
+        for (int i = 0; i < limit; i++) {
+            String v = values[i].trim();
+            parsedValues.set(i, v.isEmpty() ? defaultValue : Long.parseLong(v));
+        }
+        return parsedValues;
+    }
+
+    private static HashMap<String, DiscreteGenerator> createDiscreteFieldsMap(String cardinalities) {
+        ArrayList<Long> parsedCardinalities = parseCommaSeparatedIntegers(cardinalities, -1, 0);
+        HashMap<String, DiscreteGenerator> outputMap = new HashMap<String, DiscreteGenerator>();
+        for (int i = 0; i < parsedCardinalities.size(); i++) {
```

Comment:
nit: `for(Long value: parseCommaSeparatedIntegers(cardinalities, -1, 0))` simpler and you can lose a couple of temp variables.

Reply:
I actually need the `i` temp variable to construct the field name `"field" + i`.